EIPS/eip-7934.md

```diff
@@ -0,0 +1,77 @@
+---
+eip: 7934
+title: RLP Execution Block Size Limit
+description: Introduce a protocol-level cap on the maximum RLP-encoded block size to 10 MB, including a 512 KB margin for beacon block size.
+author: Giulio Rebuffo (@Giulio2002), Ben Adams (@benaadams), Storm Slivkoff (@sslivkoff)
+discussions-to: https://ethereum-magicians.org/t/eip-7934-add-bytesize-limit-to-blocks/23589
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-04-16
+---
+
+## Abstract
+
+This proposal introduces a protocol-level cap on the maximum RLP-encoded execution block size to 10 megabytes (MB), which includes a margin of 512 KB to account for beacon block sizes.
+
+## Motivation
+
+Currently, Ethereum does not enforce a strict upper limit on the encoded size of blocks. This lack of constraint can result in:
+
+1. **Network Instability**: Extremely large blocks slow down propagation and increase the risk of temporary forks and reorgs.
+2. **DoS Risks**: Malicious actors could generate exceptionally large blocks to disrupt network performance.
+
+Additionally, blocks exceeding 10 MB are not propagated by the consensus layer's (CL) gossip protocol, potentially causing network fragmentation or denial-of-service (DoS) conditions.
+
+By imposing a protocol-level limit on the RLP-encoded block size, Ethereum can ensure enhanced resilience against targeted attacks on block validation times. Adding an additional margin of 512 KB explicitly accommodates beacon block sizes, ensuring compatibility across network components.
+
+## Specification
+
+### Block Size Cap
+
+- Introduce constants:
+  - `MAX_BLOCK_SIZE` set to **10 MB (10,485,760 bytes)**
+  - `MARGIN` set to **512 KB (524,288 bytes)**
+  - `MAX_RLP_BLOCK_SIZE` calculated as `MAX_BLOCK_SIZE - MARGIN`
+- Any RLP-encoded block exceeding `MAX_RLP_BLOCK_SIZE` must be considered invalid.
+
+Thus add the following check to the Ethereum protocol:
+
+```python
+MAX_BLOCK_SIZE = 10_485_760  # 10 MB
+MARGIN = 524_288  # 512 KB
+MAX_RLP_BLOCK_SIZE = MAX_BLOCK_SIZE - MARGIN
```

Comment:
Can we edit the variable names?

```suggestion
GOSSIP_UPPER_LIMIT = 10_485_760  # 10 MiB
SAFETY_MARGIN = 524_288  # 512 KiB
MAX_RLP_BLOCK_SIZE = GOSSIP_UPPER_LIMIT - SAFETY_MARGIN
```

I think this is somewhat closer to what the "purpose" is of these variables (but far from perfect, would love to hear some other suggestions what could be even better here)